src/test/java/org/fogbowcloud/manager/core/processors/SpawningProcessorTest.java

```diff
@@ -93,212 +92,291 @@ public void tearDown() {
-        		SshTunnelConnectionData.class))).thenReturn(true);
-
-        Assert.assertNull(this.fulfilledOrderList.getNext());
+        Mockito.doThrow(new UnexpectedException()).when(this.spawningProcessor)
+                .processSpawningOrder(order);
 
         this.thread = new Thread(this.spawningProcessor);
         this.thread.start();
-        Thread.sleep(500);
 
-        Assert.assertNull(this.spawningOrderList.getNext());
+        Thread.sleep(500);
 
-        Order test = this.fulfilledOrderList.getNext();
-        assertNotNull(test);
-        Assert.assertEquals(order.getInstanceId(), test.getInstanceId());
-        Assert.assertEquals(OrderState.FULFILLED, test.getOrderState());
+        this.spawningProcessor.processSpawningOrder(order);
     }
+    
+    @Test
+    public void testTryingProcessComputeOrderNotSpawning() throws Exception {
+        Order order = spyComputeOrder();
+        order.setOrderState(OrderState.OPEN);
+        this.openOrderList.addItem(order);
+        
+        this.spawningProcessor.processSpawningOrder(order);
 
+        Assert.assertNotNull(this.openOrderList.getNext());
+        Assert.assertNull(this.fulfilledOrderList.getNext());
+    }
+    
     @Test
-    public void testRunProcessComputeOrderInstanceInactive() throws InterruptedException {
-        Order order = this.createOrder();
+    public void testRunProcessComputeOrderWithoutSSHConnectivity() throws Exception {
+        Order order = spyComputeOrder();
         order.setOrderState(OrderState.SPAWNING);
         this.spawningOrderList.addItem(order);
+        
+        Mockito.doReturn(false).when(order).isRequesterRemote(Mockito.anyString());
 
-        String instanceId = "fake-id";
-        ComputeInstance computeOrderInstance =
-                spy(new ComputeInstance(instanceId));
-        computeOrderInstance.setState(InstanceState.INACTIVE);
-        order.setInstanceId(instanceId);
+        Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
+        orderInstance.setState(InstanceState.READY);
+        order.setInstanceId(FAKE_INSTANCE_ID);
 
+        Mockito.doReturn(orderInstance).when(this.cloudConnector).getInstance(
+                Mockito.any(Order.class));
+
+        Mockito.when(this.tunnelingService.getExternalServiceAddresses(
+                Mockito.anyString())).thenReturn(new HashMap<>());
+        
+        Mockito.when(this.sshConnectivity.checkSSHConnectivity(
+                Mockito.any(SshTunnelConnectionData.class))).thenReturn(false);
+        
         this.thread = new Thread(this.spawningProcessor);
         this.thread.start();
         Thread.sleep(500);
 
-        Order test = this.spawningOrderList.getNext();
-        Assert.assertNotNull(test);
-        Assert.assertEquals(OrderState.SPAWNING, test.getOrderState());
+        Assert.assertNotNull(this.spawningOrderList.getNext());
+        Assert.assertNull(this.fulfilledOrderList.getNext());
```

Comment:
I'd assert that failedOrderList is also null, once the spawning will transition the order to fulfilled or failed state.

Reply:
Actually, every time that you make an assert in fulfilledOrderList, I thinkthat would be great to make an assert in failedOrderList